### ISSUE TITLE
Zebra crash when installing for some package (stable version)

### ISSUE THREAD

**Issue by udevsharold** [zbrateam/Zebra#1572]
**Describe the bug**
It crashed without any log for some package, in my case, Zefy (https://udevsharold.github.io/repo/). Either a fresh install or reinstall will crash it. Removing it will not. I'm guessing because of how Zebra handling postinst. However, when reinstalling it using Cydia and back to Zebra and reinstalling, it installs just fine. Reproducible on both iPad and iPhone.

**To Reproduce**
Install/reinstall Zefy. If it doesn't crash for the first time, do keep reinstalling and it most likely will crash on second reinstall.

**Expected behavior**
Not crash.

**Screenshots**
https://imgur.com/a/7eVEfyp

**Device Information:** 

 - iOS Version: 13.5
 - Zebra Version: 1.1.13
 - Device Type: iPhone & iPad
 - Jailbreak Type: Unc0ver

**Crash Log or System Log**
None.

**Additional context**
None.

**Comment by udevsharold** [comment]
Okay, my bad. I forgot to quote the text after `echo` in postinst. But I think Zebra should handle it more properly rather than crashing, so I'm leaving this open.

This will crash Zebra:
`echo unloading something`
This will not
`echo "unloading something"`

**Comment by udevsharold** [comment]
Update: Okay, it still crash when upgrading package.
So it's probably not because of the absent of the quotes.

**Comment by absidue** [comment]
Could you please upload your postinst to a paste site (e.g. pastebin or hastebin) and reply with the link?

**Comment by wstyres** [comment]
Would also be helpful to know if this is only occurring with this package and whether or not this occurs on the beta version of Zebra.

**Comment by udevsharold** [comment]
@absidue Here's the links,
postinst: https://pastebin.ubuntu.com/p/sxsmjnB6RT/
postrm: https://pastebin.ubuntu.com/p/XcP2tKxGrp/

@wstyres I could confirm it's reproducible on 1.2~beta6a, both install and removal.

**Comment by absidue** [comment]
Try removing the `exit 0` at the end. If the other commands produce non zero exit codes that you are happy to ignore you can use `true` instead.

**Comment by udevsharold** [comment]
Using `true` unfortunately still crash it. However, I think that's not the conventional way for both postinst/postrm either. Noticed another bug that when I try to unblock an author in beta, Zebra also crash. I should open another ticket for this.

**Comment by absidue** [comment]
After some investigation the crash seems to be caused by your `postinst` and `postrm` files killing `runningboardd`.

The little information we could find on the daemon seem to say that it's purpose is to detect errors?

**Comment by wstyres** [comment]
Looks like iOS does store crash logs. Could you go to `Settings > Privacy > Analytics & Improvements > Analytics Data` and scroll all the way down to the end and look at the "Zebra" logs? Extracting each of those would be helpful in debugging the crash. It is very likely that killing runningboard like you do is causing this crash and while one solution is just to remove the runningboard kill in your scripts we want to be sure the root cause of the crash is fixed.

**Comment by udevsharold** [comment]
@absidue not sure about its purpose is to detect errors, but from my RE during Zefy development, it handles almost all the launching/running/termination of processes in iOS. The crashing seems to be occurs only during the installation of Zefy, where Zebra handles the `postinst` or `postrm`. If you open Zebra and SSH into your device and run `killall -9 runningboardd`, there's no crash. I think this implies that somehow when Zebra itself trying to kill runningboardd, it crashed.

@wstyres removing it does 50/50 fixed so far for several tries. It's 50/50 because if you do try many times downgrading the package (the one with killing runningboardd) to debug version (the one without killing runningboardd), Zebra stills crash.It's as though as Zebra is still using the `postinst` from the higher version package. Is that the case?  You could try the debug version I uploaded on the repo, it's marked as version `1.0.0~debug`. From the crash log, it seems like it's indeed related to runningboardd/launchd:

```
{"app_name":"Zebra","timestamp":"2020-09-25 23:39:47.00 +0800","app_version":"1.2~beta6","slice_uuid":"ccb45446-d6a1-3ad3-94b3-3ccfe88a9553","adam_id":0,"build_version":"1","bundleID":"xyz.willy.Zebra","share_with_app_devs":0,"is_first_party":1,"bug_type":"109","os_version":"iPhone OS 13.5 (17F75)","incident_id":"D52D83D9-14E2-49B8-A275-FC70F45C5134","name":"Zebra"}
Incident Identifier: D52D83D9-14E2-49B8-A275-FC70F45C5134
CrashReporter Key:   10bcce77eb64fde041b45cfef141eb65a3c7a01b
Hardware Model:      iPhone10,6
Process:             Zebra [41988]
Path:                /Applications/Zebra.app/Zebra
Identifier:          xyz.willy.Zebra
Version:             1 (1.2~beta6)
Code Type:           ARM-64 (Native)
Role:                unknown
Parent Process:      launchd [1]
Coalition:           xyz.willy.Zebra [5982]


Date/Time:           2020-09-25 23:39:47.3880 [REDACTED]
Launch Time:         2020-09-25 23:39:32.2269 [REDACTED]
OS Version:          iPhone OS 13.5 (17F75)
Release Type:        User
Baseband Version:    3.05.00
Report Version:      104

Exception Type:  EXC_CRASH (SIGKILL)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: Namespace RUNNINGBOARD, Code 0xdead10cc
Triggered by Thread:  0

Thread 0 name:  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x00000001848a9198 0x1848a5000 + 16792
1   libsystem_kernel.dylib        	0x00000001848a860c 0x1848a5000 + 13836
2   CoreFoundation                	0x0000000184a53468 0x1849ab000 + 689256
3   CoreFoundation                	0x0000000184a4e49c 0x1849ab000 + 668828
4   CoreFoundation                	0x0000000184a4dce8 0x1849ab000 + 666856
5   GraphicsServices              	0x000000018eb9838c 0x18eb95000 + 13196
6   UIKitCore                     	0x0000000188b7c444 0x18814e000 + 10675268
7   Zebra                         	0x0000000102c9babc 0x102c6c000 + 195260
8   libdyld.dylib                 	0x00000001848d58f0 0x1848d4000 + 6384

Thread 1:
0   libsystem_pthread.dylib       	0x00000001847ef738 0x1847e1000 + 59192

Thread 2:
0   libsystem_pthread.dylib       	0x00000001847ef738 0x1847e1000 + 59192

Thread 3 name:  com.apple.uikit.eventfetch-thread
Thread 3:
0   libsystem_kernel.dylib        	0x00000001848a9198 0x1848a5000 + 16792
1   libsystem_kernel.dylib        	0x00000001848a860c 0x1848a5000 + 13836
2   CoreFoundation                	0x0000000184a53468 0x1849ab000 + 689256
3   CoreFoundation                	0x0000000184a4e49c 0x1849ab000 + 668828
4   CoreFoundation                	0x0000000184a4dce8 0x1849ab000 + 666856
5   Foundation                    	0x0000000184d9101c 0x184d89000 + 32796
6   Foundation                    	0x0000000184d90efc 0x184d89000 + 32508
7   UIKitCore                     	0x0000000188c1f5dc 0x18814e000 + 11343324
8   Foundation                    	0x0000000184ebfe20 0x184d89000 + 1273376
9   libsystem_pthread.dylib       	0x00000001847ebd98 0x1847e1000 + 44440
10  libsystem_pthread.dylib       	0x00000001847ef74c 0x1847e1000 + 59212

Thread 4:
0   libsystem_pthread.dylib       	0x00000001847ef738 0x1847e1000 + 59192

Thread 5:
0   libsystem_pthread.dylib       	0x00000001847ef738 0x1847e1000 + 59192

Thread 6:
0   libsystem_pthread.dylib       	0x00000001847ef738 0x1847e1000 + 59192

Thread 7 name:  Dispatch queue: com.apple.root.user-initiated-qos
Thread 7:
0   libsystem_kernel.dylib        	0x00000001848cb104 0x1848a5000 + 155908
1   libsystem_c.dylib             	0x0000000184733748 0x1846c4000 + 456520
2   libsystem_c.dylib             	0x0000000184733618 0x1846c4000 + 456216
3   libdispatch.dylib             	0x000000018479a9a8 0x184740000 + 371112
4   libdispatch.dylib             	0x000000018479b524 0x184740000 + 374052
5   libdispatch.dylib             	0x000000018478165c 0x184740000 + 267868
6   libdispatch.dylib             	0x0000000184781cd0 0x184740000 + 269520
7   libsystem_pthread.dylib       	0x00000001847ecb38 0x1847e1000 + 47928
8   libsystem_pthread.dylib       	0x00000001847ef740 0x1847e1000 + 59200

Thread 8 name:  com.google.firebase.crashlytics.MachExceptionServer
Thread 8:
0   libsystem_kernel.dylib        	0x00000001848a9198 0x1848a5000 + 16792
1   libsystem_kernel.dylib        	0x00000001848a860c 0x1848a5000 + 13836
2   Zebra                         	0x0000000102decfc4 0x102c6c000 + 1576900
3   libsystem_pthread.dylib       	0x00000001847ebd98 0x1847e1000 + 44440
4   libsystem_pthread.dylib       	0x00000001847ef74c 0x1847e1000 + 59212

Thread 9 name:  com.apple.NSURLConnectionLoader
Thread 9:
0   libsystem_kernel.dylib        	0x00000001848a9198 0x1848a5000 + 16792
1   libsystem_kernel.dylib        	0x00000001848a860c 0x1848a5000 + 13836
2   CoreFoundation                	0x0000000184a53468 0x1849ab000 + 689256
3   CoreFoundation                	0x0000000184a4e49c 0x1849ab000 + 668828
4   CoreFoundation                	0x0000000184a4dce8 0x1849ab000 + 666856
5   CFNetwork                     	0x0000000187d0e894 0x187d0d000 + 6292
6   Foundation                    	0x0000000184ebfe20 0x184d89000 + 1273376
7   libsystem_pthread.dylib       	0x00000001847ebd98 0x1847e1000 + 44440
8   libsystem_pthread.dylib       	0x00000001847ef74c 0x1847e1000 + 59212

Thread 10 name:  com.apple.CFSocket.private
Thread 10:
0   libsystem_kernel.dylib        	0x00000001848cb014 0x1848a5000 + 155668
1   CoreFoundation                	0x0000000184a61134 0x1849ab000 + 745780
2   libsystem_pthread.dylib       	0x00000001847ebd98 0x1847e1000 + 44440
3   libsystem_pthread.dylib       	0x00000001847ef74c 0x1847e1000 + 59212

Thread 11 name:  AVAudioSession Notify Thread
Thread 11:
0   libsystem_kernel.dylib        	0x00000001848a9198 0x1848a5000 + 16792
1   libsystem_kernel.dylib        	0x00000001848a860c 0x1848a5000 + 13836
2   CoreFoundation                	0x0000000184a53468 0x1849ab000 + 689256
3   CoreFoundation                	0x0000000184a4e49c 0x1849ab000 + 668828
4   CoreFoundation                	0x0000000184a4dce8 0x1849ab000 + 666856
5   AVFAudio                      	0x00000001917f7a2c 0x191792000 + 416300
6   AVFAudio                      	0x00000001918487a0 0x191792000 + 747424
7   libsystem_pthread.dylib       	0x00000001847ebd98 0x1847e1000 + 44440
8   libsystem_pthread.dylib       	0x00000001847ef74c 0x1847e1000 + 59212

Thread 12:
0   libsystem_pthread.dylib       	0x00000001847ef738 0x1847e1000 + 59192

Thread 13:
0   libobjc.A.dylib               	0x00000001847f63dc 0x1847f2000 + 17372
1   CoreFoundation                	0x0000000184a54c9c 0x1849ab000 + 695452
2   CoreFoundation                	0x0000000184a54c9c 0x1849ab000 + 695452
3   CoreFoundation                	0x0000000184a7405c 0x1849ab000 + 823388
4   CoreFoundation                	0x0000000184a74554 0x1849ab000 + 824660
5   Foundation                    	0x0000000184d8b5a4 0x184d89000 + 9636
6   Zebra                         	0x0000000102d051b0 0x102c6c000 + 627120
7   Zebra                         	0x0000000102cfb2d8 0x102c6c000 + 586456
8   Zebra                         	0x0000000102cfc4c0 0x102c6c000 + 591040
9   Zebra                         	0x0000000102cfc454 0x102c6c000 + 590932
10  Zebra                         	0x0000000102cf4d14 0x102c6c000 + 560404
11  Zebra                         	0x0000000102cf488c 0x102c6c000 + 559244
12  Zebra                         	0x0000000102c7a114 0x102c6c000 + 57620
13  Zebra                         	0x0000000102c793e4 0x102c6c000 + 54244
14  Foundation                    	0x0000000184ebfe20 0x184d89000 + 1273376
15  libsystem_pthread.dylib       	0x00000001847ebd98 0x1847e1000 + 44440
16  libsystem_pthread.dylib       	0x00000001847ef74c 0x1847e1000 + 59212

Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000010004005   x1: 0x0000000007000806   x2: 0x0000000000000000   x3: 0x0000000000000c00
    x4: 0x0000000000001c03   x5: 0x00000000ffffffff   x6: 0x0000000000000000   x7: 0x000000000000000c
    x8: 0x00000000fffffbbf   x9: 0x0000000007000906  x10: 0x01a84b991ed74c08  x11: 0x000000b2f4dc70cc
   x12: 0x00000000016e3600  x13: 0x0000000000025ef6  x14: 0x0000000000000e00  x15: 0x00000e0000000000
   x16: 0xffffffffffffffe1  x17: 0x0000000000000001  x18: 0x0000000000000000  x19: 0x0000000000000000
   x20: 0x00000000ffffffff  x21: 0x0000000000001c03  x22: 0x0000000000000c00  x23: 0x000000016d192e30
   x24: 0x0000000007000806  x25: 0x0000000000000000  x26: 0x0000000007000806  x27: 0x0000000000000c00
   x28: 0x0000000000000001   fp: 0x000000016d192d20   lr: 0x00000001848a860c
    sp: 0x000000016d192cd0   pc: 0x00000001848a9198 cpsr: 0x60000000
   esr: 0x56000080  Address size fault

```

**Comment by wstyres** [comment]
This crash might be hard to track down. If it still crashes when removing the call to kill runningboard, could you post a crash log when that occurs?

**Comment by wstyres** [comment]
Have you noticed any further crashes to this? I haven't really found anything related to runningboard when looking so I'm not sure the source of the crashes.

**Comment by udevsharold** [comment]
I currently still staying at beta6 due to this issue: https://github.com/zbrateam/Zebra/issues/1657
However, I haven't notice this issue with anymore in beta6. The issue seems like to just go away, I'm not really sure why. I think we should close this issue for now, and only reopen it should this arise again.
